src/flows/context/flow.query.tsx

```diff
@@ -213,10 +213,20 @@ export const FlowQueryProvider: FC = ({children}) => {
   }, [flow])
 
   const generateMap = (): Stage[] => {
-    if (
-      !_map.current ||
-      (!_map.current.length && (flow?.data?.allIDs ?? []).length)
-    ) {
+    // this is to get around an issue where a panel is added, which triggers the useEffect that updates
+    // _map.current and a rerender that updates the panel view components within the same render cycle
+    // leading to a panel on the list without a corresponding map entry
+    const forceUpdate =
```

Comment:
Could we memoize this so we don't repeat it every time if the panels are unchanged? Seems like it could introduce a lot of overhead of there are a lot of panels

Reply:
thankfully the overhead of the sort and string match was minimal up to the 100 panels / notebook i was testing at (i think the prod median is 4, but our numbers are real shoddy around that), especially compared to the parse -> format_from_js_file cycle this is replacing. storing the result sets for visualization is what starts crippling the ui at that point (running out of ram), so i didn't go much further. tying into the pr text, this is to shore us up for a bit while looking for a better solution. like.. there's an implicit coupling across abstractions between the list of panels, and the queries they generate, which suggests that the abstraction is wrong and by solving that, we should be able to remove the matching all together. but while typing all this out... we dont need to sort the list of ids, because it's already a sorted list, so the overhead then just becomes a string match.  lemme push that up